packages/ios_platform_images/lib/ios_platform_images.dart

```diff
@@ -8,38 +8,72 @@ import 'package:flutter/painting.dart';
 import 'package:flutter/foundation.dart'
     show SynchronousFuture, describeIdentity;
 
+class _FutureImageStreamCompleter extends ImageStreamCompleter {
+  final Future<double> futureScale;
+  final InformationCollector informationCollector;
+
+  _FutureImageStreamCompleter(
+      {Future<ui.Codec> codec, this.futureScale, this.informationCollector})
+      : assert(codec != null),
+        assert(futureScale != null) {
+    codec.then<void>(_onCodecReady, onError: (dynamic error, StackTrace stack) {
+      reportError(
+        context: ErrorDescription('resolving a single-frame image stream'),
+        exception: error,
+        stack: stack,
+        informationCollector: informationCollector,
+        silent: true,
+      );
+    });
+  }
+
+  Future<void> _onCodecReady(ui.Codec codec) async {
+    try {
+      ui.FrameInfo nextFrame = await codec.getNextFrame();
+      double scale = await futureScale;
+      setImage(ImageInfo(image: nextFrame.image, scale: scale));
+    } catch (exception, stack) {
+      reportError(
+        context: ErrorDescription('resolving an image frame'),
+        exception: exception,
+        stack: stack,
+        informationCollector: this.informationCollector,
+        silent: true,
+      );
+    }
+  }
+}
+
 /// Performs exactly like a [MemoryImage] but instead of taking in bytes it takes
 /// in a future that represents bytes.
-class FutureMemoryImage extends ImageProvider<FutureMemoryImage> {
+class _FutureMemoryImage extends ImageProvider<_FutureMemoryImage> {
   /// Constructor for FutureMemoryImage.  [_futureBytes] is the bytes that will
```

Comment:
Since this class is private you probably don't need Dartdoc comments here.

If you do feel compelled to document `_futureBytes` and `_futureScale` you should do that where they're defined. (First Dartdoc paragraph should always be a single sentence.)

Reply:
I already went through the trouble documenting it.  I know they aren't required now, but they are still useful.